src/View/Helper/HtmlHelper.php

```diff
@@ -219,7 +219,11 @@ public function meta($type, $content = null, array $options = []): ?string
         $out = '';
 
         if (isset($options['link'])) {
-            $options['link'] = $this->Url->assetUrl($options['link']);
+            if (is_array($options['link'])) {
+                $options['link'] = $this->Url->build($options['link']);
```

Comment:
Shouldnt the array case also run into assetUrl() to append the asset part? Someone just provided the link as non-string form yet.
That hasnt been the case so far either, from what I read. But maybe it should be this way for completeness/consistency?
```php
if (is_array($options['link'])) {
    $url = $this->Url->build($options['link']);
    $options['link'] = $this->Url->assetUrl($url);
```
etc?

Reply:
The only additional thing that `assetUrl()` does after generating a string URL is check and add timestamp to query string. If the URL is an array then it's for a controller in which case timestamp check becomes redundant.

Reply:
Also if you see the diff, earlier `assetUrl()` simply returned the output of `build()` method without doing anything extra.

Reply:
I know ("That hasnt been the case so far either"). But it makes sense to not have to be done for controller ones I guess. Or you have to do it manually if you serve assets this way :)